### ISSUE TITLE
Cannot read property 'add' of null (Failed to activate the file-icons package)

### ISSUE THREAD

**Issue by tjgerot** [file-icons/atom#524]
It started happening after the latest update of this package when I pressed Ctrl+T to open fuzzy finder. Uninstalled the package and reinstalled and it tried to implement it's logos on my file tree and it failed with this error.

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Microsoft Windows 10 Home
**Thrown From**: [file-icons](https://github.com/file-icons/atom) package, v2.0.10

### Stack Trace

Failed to activate the file-icons package

```
At Cannot read property 'add' of null

TypeError: Cannot read property 'add' of null
	at Nuclide.init (C:\Users\Thomas\.atom\packages\file-icons\lib\consumers\consumer.js:49:19)
	at ConsumerManager.init (C:\Users\Thomas\.atom\packages\file-icons\lib\consumers\all.js:55:13)
	at Object.activate (C:\Users\Thomas\.atom\packages\file-icons\lib\main.js:19:13)
	at Package.module.exports.Package.activateNow (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package.js:191:19)
	at C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package.js:164:32
	at Package.module.exports.Package.measure (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package.js:94:15)
	at C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package.js:157:26
	at Package.module.exports.Package.activate (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package.js:154:34)
	at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\package-manager.js:550:34)
	at C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:439:29
	at exit (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:73:16)
	at triggerExitCallback (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\buffered-process.js:299:11)
	at ChildProcess.<anonymous> (C:\Users\Thomas\AppData\Local\atom\app-1.13.0\resources\app.asar\src\buffered-process.js:329:11)
	at emitTwo (events.js:106:13)
	at ChildProcess.emit (events.js:191:7)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
```

### Commands

```
     -4:46.8.0 core:backspace (input.hidden-input)
     -4:40.8.0 core:save (input.hidden-input)
     -1:16.7.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
     -1:14.9.0 editor:consolidate-selections (input.hidden-input)
     -1:14.9.0 core:cancel (input.hidden-input)
     -1:14.4.0 core:undo (input.hidden-input)
     -1:13.9.0 core:save (input.hidden-input)
  2x -1:13.1.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
     -1:03.7.0 command-palette:toggle (input.hidden-input)
     -1:02.4.0 core:backspace (input.hidden-input)
     -1:00.8.0 core:confirm (input.hidden-input)
     -1:00.8.0 settings-view:view-installed-packages (input.hidden-input)
     -0:59.5.0 editor:consolidate-selections (input.hidden-input)
     -0:59.5.0 core:cancel (input.hidden-input)
  5x -0:26.5.0 core:backspace (input.hidden-input)
     -0:17.9.0 core:confirm (input.hidden-input)
```

### Config

~~~json
{
	"core": {
		"audioBeep": false,
		"autoHideMenuBar": true,
		"openEmptyEditorOnStart": false,
		"projectHome": "C:\\Users\\Thomas\\Documents",
		"telemetryConsent": "limited"
	}
}
~~~

### Packages

~~~bash
# Enabled Themes
one-dark-syntax      1.6.0
one-dark-ui          1.8.2

# Consumer packages
archive-view         0.62.0
find-and-replace     0.204.5
fuzzy-finder         1.4.0
tabs                 0.103.1
tree-view            0.211.1

# Active (Core excluded)
atom-beautify        0.29.16
atom-clock           0.1.3
autoclose-html       0.23.0
autocomplete-python  1.8.26
git-diff             1.2.0
hex                  0.6.2
language-batch       0.3.3
merge-conflicts      1.4.4
pretty-json          1.6.1
rest-client          1.3.1
wakatime             6.0.12

# Inactive
autocomplete-java    1.2.5
file-icons           2.0.10
KeyCount             0.1.1
link                 0.31.2
open-on-github       1.2.1
package-generator    1.0.2
script               3.14.0
todo-show            1.9.0
youtube-pane         2.0.0
~~~

**Comment by Alhadis** [comment]
"Started happening"? How many times has this happened, exactly?

Just so you know, when things like this are happening after an update, your best bet is to reload Atom, not uninstall and reinstall. In fact, in this package's case, reinstalling in the same workspace session is actually likely to make things worse.

**Comment by tellkonkle** [comment]
This is happening to me to after the last update (installed about 30 minutes ago). I appears to only happen when the file tree refreshes after a file is created by an external program.

**Comment by Alhadis** [comment]
So, is this being displayed *repeatedly?* Are you still running the same session right now as you were when you installed the package?

I'm starting to think I should display a notification to users who're installing this, clearly underlining the need for a restart...

(I also hope the Atom team decide to merge atom/tree-view#1009, because this tree-view hackery can't go on forever)

**Comment by Alhadis** [comment]
@tellkonkle In future, please always include the operating system you're running. It helps a lot.

**Comment by tellkonkle** [comment]
Sorry about that. Running on Windows 7 (Home Premium) 64-bit. The file icon problem has persisted even after a system reboot. The exception will display repeatedly each time the file tree is refreshed with a new file. So, for example, when I'm debugging some PHP code that puts a dump in several log files that show up in my file tree, the exception will show several times each time the code is ran and a new log is added.

**Comment by tjgerot** [comment]
@Alhadis I had reloaded Atom. I previously edited parts of the package's local source and reinstalled the package to ensure the errors were unrelated to my changes. Also, I don't believe the reinstallation of the package should make it any more likely to break as I had cleanly removed the package and all references to it.

I had disabled the extension after opening this issue and re-enabled it just now (still running 2.0.10) and all my icons rendered correctly, without any errors. I'm still in the same (Meteor.js) project with the same files, so I doubt it is related to the files open.

I initially thought the fix was a result of a system reboot or completely killing and restarting the Atom process, however @tellkonkle's report seems to throw that out. I'd be interested to see @tellkonkle's Atom-generated debug log.

**Comment by Alhadis** [comment]
> I previously edited parts of the package's local source

Yeah, okay. Don't ever do that. Completely voids warranty, so to speak.

**Comment by despairblue** [comment]
Happened to me after upgrading to `2.0.11` from within atom and reloading atom `Ctrl-Meta-Cmd-L`|`Ctrl-Alt-R`.

Restarting doesn't help either. I'll try to reinstall the package.

--- 
<details>
 <summary>Crash Report</summary>

**Atom Version**: 1.13.0
**Electron Version**: 1.3.13
**System**: Mac OS X 10.12.2
**Thrown From**: [file-icons](https://github.com/file-icons/atom) package, v2.0.11


### Stack Trace

Failed to activate the file-icons package

```
At Cannot read property 'atom.file-icons' of undefined

TypeError: Cannot read property 'atom.file-icons' of undefined
    at ConsumerManager.needsPatch (/Users/despairblue/.atom/packages/file-icons/lib/consumers/all.js:50:43)
    at new ConsumerManager (/Users/despairblue/.atom/packages/file-icons/lib/consumers/all.js:23:12)
    at Object.<anonymous> (/Users/despairblue/.atom/packages/file-icons/lib/consumers/all.js:79:18)
    at Module._compile (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:109:30)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app.asar/src/compile-cache.js:216:21)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/Users/despairblue/.atom/packages/file-icons/lib/main.js:6:24)
    at Module._compile (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:109:30)
    at Object.value [as .js] (/Applications/Atom.app/Contents/Resources/app.asar/src/compile-cache.js:216:21)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
    at Module.require (module.js:483:17)
    at require (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:766:27)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:181:16)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:164:32
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:94:15)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package.js:157:26
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package.js:154:34)
    at PackageManager.module.exports.PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:550:34)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:531:29
    at Config.module.exports.Config.transactAsync (/Applications/Atom.app/Contents/Resources/app.asar/src/config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:526:19)
    at PackageManager.module.exports.PackageManager.activate (/Applications/Atom.app/Contents/Resources/app.asar/src/package-manager.js:508:46)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/atom-environment.js:831:28
```

### Commands

```
  4x -0:00.3.0 blur (atom-pane.pane.active)
     -0:00.2.0 window:toggle-dev-tools (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "closeEmptyWindows": false,
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig",
        ".hgrc"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "atom-color-highlight",
      "atomatigit",
      "es-navigation",
      "filetype-color",
      "grunt-runner",
      "incremental-search",
      "linter-scspell",
      "markdown-scroll-sync",
      "metrics",
      "minimap-color-highlight",
      "minimap-find-and-replace",
      "minimap-git-diff",
      "minimap-highlight-selected",
      "minimap",
      "navigate",
      "tab-smart-sort",
      "Tern",
      "test-status",
      "themed-settings",
      "travis-ci-status",
      "tree-view-breadcrumb",
      "tree-view-open-files",
      "welcome",
      "linter-jshint",
      "markdown-preview-opener",
      "linter-htmlhint",
      "atom-pair",
      "Zen",
      "angularjs",
      "atom-zeal",
      "firepad",
      "language-tss",
      "lazy-motion",
      "open-plus",
      "swap-panes",
      "json-schema",
      "javascript-snippets",
      "linter-js-standard",
      "starred-package-installer",
      "linter-puppet-lint",
      "move-panes",
      "imdone-atom",
      "method-separator",
      "pane-layout-plus",
      "pane-move-plus",
      "git-blame",
      "git-log",
      "gorename",
      "language-haskell",
      "language-puppet",
      "navigator-godef",
      "tester-go",
      "grammar-token-limit",
      "block-travel",
      "linter-js-standard-engine",
      "react",
      "goto",
      "vim-mode",
      "compare-files",
      "vim-mode-clipboard-plus",
      "clipboard-plus",
      "atom-alignment",
      "date",
      "expose",
      "git-control",
      "go-debug",
      "godoc",
      "haskell-grammar",
      "language-api-blueprint",
      "language-docker",
      "language-dot",
      "language-env",
      "language-latex",
      "language-nimrod",
      "language-thrift",
      "latex",
      "markdown-pdf",
      "stacktrace",
      "todo-show",
      "language-clojure",
      "language-csharp",
      "language-java",
      "language-make",
      "language-mustache",
      "language-objective-c",
      "language-perl",
      "language-php",
      "language-property-list",
      "language-ruby",
      "language-ruby-on-rails",
      "language-sql",
      "linter-markdown",
      "git-history",
      "gofmt",
      "autocomplete-go",
      "builder-go",
      "go-config",
      "go-get",
      "go-plus",
      "gometalinter-linter",
      "maximize-panes",
      "linter-elixirc",
      "autocomplete-elixir",
      "api-blueprint-preview",
      "linter-api-blueprint",
      "environment",
      "ex-mode",
      "git-difftool",
      "git-time-machine",
      "incsearch",
      "navigator-go",
      "symbols-tree-view",
      "vim-surround",
      "tool-bar",
      "tree-view",
      "gitignore-snippets",
      "highlight-selected",
      "autocomplete-snippets",
      "autocomplete-python",
      "kite",
      "language-python"
    ],
    "excludeVcsIgnoredPaths": false,
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "elm-stuff",
      "node_modules"
    ],
    "projectHome": "~/vcs/git/"
  },
  "file-icons": {
    "coloured": false
  }
}
```

### Installed Packages

```coffee
# User
advanced-open-file, v0.16.5 (inactive)
atom-beautify, v0.29.16 (inactive)
atom-debugger, v0.1.6 (inactive)
atom-fuzzy-grep, v0.15.0 (inactive)
atom-typescript, v10.1.13 (inactive)
atomic-chrome, v0.3.0 (inactive)
autocomplete-elixir, v1.5.14 (inactive)
autocomplete-emojis, v2.5.0 (inactive)
autocomplete-go, v1.3.0 (inactive)
autocomplete-paths, v1.0.5 (inactive)
autocomplete-python, v1.8.27 (inactive)
blame, v0.10.2 (inactive)
bug-report, v0.7.1 (inactive)
character-table, v0.4.3 (inactive)
choose-pane, v0.5.1 (inactive)
color-picker, v2.2.4 (inactive)
dash, v1.7.0 (inactive)
docblockr, v0.9.1 (inactive)
editorconfig, v2.2.0 (inactive)
elm-format, v2.1.0 (inactive)
elmjutsu, v3.1.0 (inactive)
file-icons, v2.0.11 (inactive)
git-diff-popup, v0.1.8 (inactive)
git-plus, v7.2.0 (inactive)
gitignore-snippets, v0.2.5 (inactive)
grammar-token-limit, v0.1.1 (inactive)
graphviz-preview, v1.7.0 (inactive)
haskell-grammar, v0.4.0 (inactive)
highlight-selected, v0.12.0 (inactive)
html-to-elm, v0.2.0 (inactive)
hyperclick, v0.0.40 (inactive)
indentation-indicator, v1.0.0 (inactive)
js-hyperclick, v1.10.1 (inactive)
jumpy, v3.1.3 (inactive)
keybinding-cheatsheet, v0.1.1 (inactive)
kite, v0.6.7 (inactive)
language-babel, v2.49.4 (inactive)
language-docker, v1.1.7 (inactive)
language-dot, v1.6.0 (inactive)
language-dots, v0.0.4 (inactive)
language-elixir, v0.16.0 (inactive)
language-elm, v1.5.0 (inactive)
language-env, v0.2.0 (inactive)
language-fish-shell, v1.0.5 (inactive)
language-graphql, v0.8.0 (inactive)
language-haskell, v1.10.4 (inactive)
language-ini, v1.17.0 (inactive)
language-latex, v1.0.0 (inactive)
language-lua, v0.9.11 (inactive)
language-nimrod, v0.1.0 (inactive)
language-ocaml, v1.1.2 (inactive)
language-puppet, v0.22.0 (inactive)
language-rust, v0.4.9 (inactive)
language-swift, v0.5.0 (inactive)
language-thrift, v1.0.2 (inactive)
linter, v1.11.21 (inactive)
linter-alex, v3.2.0 (inactive)
linter-coffeelint, v1.2.1 (inactive)
linter-elixir-credo, v1.1.0 (inactive)
linter-elixirc, v1.1.11 (inactive)
linter-elm-make, v0.19.0 (inactive)
linter-eslint, v8.0.0 (inactive)
linter-flow, v5.5.0 (inactive)
linter-js-yaml, v1.2.6 (inactive)
linter-markdown, v3.0.2 (inactive)
linter-rust, v0.7.2 (inactive)
linter-shellcheck, v1.4.3 (inactive)
markdown-pdf, v1.5.0 (inactive)
markdown-preview-plus, v2.4.0 (inactive)
maximize-panes, v0.2.0 (inactive)
merge-conflicts, v1.4.4 (inactive)
narrow, v0.10.1 (inactive)
nix, v2.1.0 (inactive)
nuclide, v0.201.0 (inactive)
nuclide-format-js, v0.0.36 (inactive)
ocaml-indent, v0.5.2 (inactive)
ocaml-merlin, v0.9.5 (inactive)
one-tab, v0.10.0 (inactive)
open-on-bitbucket, v0.4.0 (inactive)
open-this, v0.3.0 (inactive)
package-cop, v0.2.10 (inactive)
package-sync, v1.1.0 (inactive)
pdf-view, v0.54.0 (inactive)
pigments, v0.38.0 (inactive)
prettier-atom, v0.0.10 (inactive)
prettier-eslint-atom, v2.0.1 (inactive)
project-folder, v1.2.1 (inactive)
racer, v0.20.0 (inactive)
regex-railroad-diagram, v0.16.0 (inactive)
rustfmt, v0.1.3 (inactive)
set-syntax, v0.3.2 (inactive)
sort-lines, v0.14.0 (inactive)
standard-formatter, v2.8.0 (inactive)
standardjs-snippets, v2.5.0 (inactive)
tabs-to-spaces, v1.0.3 (inactive)
tool-bar, v1.0.1 (inactive)
vim-mode-plus, v0.80.0 (inactive)
vim-mode-plus-ex-mode, v0.8.1 (inactive)
vim-mode-plus-project-find-from-search, v0.2.0 (inactive)
wakatime, v6.0.12 (inactive)
atom-dark-syntax, v0.28.0 (inactive)
atom-dark-ui, v0.53.0 (inactive)
atom-light-syntax, v0.29.0 (inactive)
atom-light-ui, v0.46.0 (inactive)
base16-tomorrow-dark-theme, v1.4.0 (inactive)
base16-tomorrow-light-theme, v1.4.0 (inactive)
one-dark-ui, v1.8.2 (inactive)
one-light-ui, v1.8.2 (inactive)
one-dark-syntax, v1.6.0 (inactive)
one-light-syntax, v1.6.0 (inactive)
solarized-dark-syntax, v1.1.1 (inactive)
solarized-light-syntax, v1.1.1 (inactive)
about, v1.7.2 (inactive)
archive-view, v0.62.0 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.14.1 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.33.1 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.29.0 (inactive)
autosave, v0.23.2 (inactive)
background-tips, v0.26.1 (inactive)
bookmarks, v0.43.2 (inactive)
bracket-matcher, v0.85.1 (inactive)
command-palette, v0.39.1 (inactive)
deprecation-cop, v0.55.1 (inactive)
dev-live-reload, v0.47.0 (inactive)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.40.0 (inactive)
find-and-replace, v0.204.5 (inactive)
fuzzy-finder, v1.4.0 (inactive)
git-diff, v1.2.0 (inactive)
go-to-line, v0.31.2 (inactive)
grammar-selector, v0.48.2 (inactive)
image-view, v0.60.0 (inactive)
incompatible-packages, v0.26.1 (inactive)
keybinding-resolver, v0.35.0 (inactive)
line-ending-selector, v0.5.1 (inactive)
link, v0.31.2 (inactive)
markdown-preview, v0.159.1 (inactive)
metrics, v1.1.2 (inactive)
notifications, v0.65.1 (inactive)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.2 (inactive)
settings-view, v0.244.0 (inactive)
snippets, v1.0.4 (inactive)
spell-check, v0.68.5 (inactive)
status-bar, v1.6.0 (inactive)
styleguide, v0.48.0 (inactive)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.1 (inactive)
timecop, v0.33.2 (inactive)
tree-view, v0.211.1 (inactive)
update-package-dependencies, v0.10.0 (inactive)
welcome, v0.35.1 (inactive)
whitespace, v0.35.0 (inactive)
wrap-guide, v0.39.0 (inactive)
language-c, v0.54.0 (inactive)
language-clojure, v0.22.1 (inactive)
language-coffee-script, v0.48.1 (inactive)
language-csharp, v0.13.0 (inactive)
language-css, v0.40.1 (inactive)
language-gfm, v0.88.0 (inactive)
language-git, v0.15.0 (inactive)
language-go, v0.43.0 (inactive)
language-html, v0.47.1 (inactive)
language-hyperlink, v0.16.1 (inactive)
language-java, v0.24.0 (inactive)
language-javascript, v0.122.0 (inactive)
language-json, v0.18.3 (inactive)
language-less, v0.29.6 (inactive)
language-make, v0.22.2 (inactive)
language-mustache, v0.13.0 (inactive)
language-objective-c, v0.15.1 (inactive)
language-perl, v0.37.0 (inactive)
language-php, v0.37.3 (inactive)
language-property-list, v0.8.0 (inactive)
language-python, v0.45.1 (inactive)
language-ruby, v0.70.2 (inactive)
language-ruby-on-rails, v0.25.1 (inactive)
language-sass, v0.57.0 (inactive)
language-shellscript, v0.23.0 (inactive)
language-source, v0.9.0 (inactive)
language-sql, v0.25.0 (inactive)
language-text, v0.7.1 (inactive)
language-todo, v0.29.1 (inactive)
language-toml, v0.18.1 (inactive)
language-xml, v0.34.12 (inactive)
language-yaml, v0.27.1 (inactive)

# Dev
No dev packages
```
</details>

**Comment by despairblue** [comment]
Uninstalled and reinstalled from the console while atom was closed and it still happens :/

**Comment by leonfeng** [comment]
Same here. The package started throwing errors after upgrading, and I never touched its local resources.

**Comment by Alhadis** [comment]
I'm a fucking idiot. Sorry guys.

Fixed in v2.0.12

**Comment by leonfeng** [comment]
No worries! Thanks for the quick fix :)

**Comment by despairblue** [comment]
Yeah, thanks for the quick fix and for working on this :)

**Comment by n0ne** [comment]
Not working for me )-:
file-icons 2.0.12, Atom 1.13.0

**Comment by tiramiseb** [comment]
I have this problem too. Atom 1.13.0, file-icons 2.0.12, Ubuntu 16.10.
Happened when disabling and re-enabling the package, or uninstalling and reinstalling it.
When closing and reopening Atom, the package loads without any error.

**Comment by Alhadis** [comment]
Resolved by a5f4e52a. You'll notice this only one more time in the next update, but it shouldn't plague you anymore afterward.
